### PR TITLE
Disable network interface

### PR DESCRIPTION
**Pull request by DanielGuramulta**
Feature mentioned in issue: #362

### CODE REVIEW

File: sdk/include/aditof/device_construction_data.h [hunk 1]
```diff
@@ -44,9 +44,11 @@ namespace aditof {
  * @brief Provides the types of sensor assosiated with the device
  */
 enum class DeviceType {
-    LOCAL,    //!< on the target
-    USB,      //!< connects to target via USB
+    LOCAL, //!< on the target
+    USB,   //!< connects to target via USB
+#ifdef HAS_NETWORK
```
Comment (dNechita) [inline sdk/include/aditof/device_construction_data.h:49]: API shouldn't change based on some build option.

File: sdk/src/device_enumerator_ethernet.cpp [hunk 2]
```diff
@@ -30,15 +30,18 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (dNechita) [inline sdk/src/device_enumerator_ethernet.cpp:30]: device_enumerator_ethernet.h/cpp could be excluded by cmake when building without Network support.

File: sdk/src/device_factory.cpp [hunk 3]
```diff
@@ -45,9 +49,11 @@ DeviceFactory::buildDevice(const aditof::DeviceConstructionData &data) {
     case DeviceType::USB: {
         return std::unique_ptr<DeviceInterface>(new UsbDevice(data));
     }
+#ifdef HAS_NETWORK
     case DeviceType::ETHERNET: {
         return std::unique_ptr<DeviceInterface>(new EthernetDevice(data));
```
Comment (dNechita) [inline sdk/src/device_factory.cpp:54]: Should report at runtime that network support is disabled.

**Comment by DanielGuramulta** [comment]
@dNechita I have addressed all the issues you mentioned and fixed them:
1) The api is no longer changed(`findDevices` method and `DeviceType` enum)
2) Moved the error message that should be seen at runtime in the systems implementation. How this works: when the user calls `getCameraAtIp(...)` and the sdk was built with `WITH_NETWORK=off` an error message will be logged telling the user that the network interface is disabled and to rebuild with the correct options enabled.

Thank you!

**Pull request merged**